aws/resource_aws_ses_receipt_filter.go

```diff
@@ -75,21 +93,33 @@ func resourceAwsSesReceiptFilterRead(d *schema.ResourceData, meta interface{}) e
 		return err
 	}
 
-	found := false
-	for _, element := range response.Filters {
```

Comment:
Since the SES API does not provide the ability to fetch a single filter, this looping is unfortunately necessary to handle accounts with multiple Receipt Filters. Otherwise it will always return errors like:

```
    TestAccAWSSESReceiptFilter_basic: testing.go:684: Step 0 error: errors during apply:

        Error: Provider produced inconsistent result after apply

        When applying changes to aws_ses_receipt_filter.test, provider "aws" produced
        an unexpected new value for was present, but now absent.

        This is a bug in the provider, which should be reported in the provider's own
        issue tracker.
```

You can use a pattern like this to move the filter object up:

```go
var filter 

for _, responseFilter := range response.Filters {
  if aws.StringValue(responseFilter.Name) == d.Id() {
    filter = responseFilter
    break
  }
}

if filter == nil {
  // bye bye
}
```

We should probably add a new covering test for this while we're adjusting this logic, e.g. a test configuration utilizing count and `count.index`, since this resource does especially need to handle multiple resources correctly.

Reply:
ive made a small change, let me know what you think. i can revert this to the original for now with minor changes